ledger/testing/randomAccounts.go

```diff
@@ -58,47 +56,62 @@ func RandomNote() []byte {
 	return note[:]
 }
 
-// RandomAccountData generates a random AccountData
+// RandomAccountData generates a random AccountData with no associated resources.
 func RandomAccountData(rewardsBase uint64) basics.AccountData {
 	var data basics.AccountData
 
 	// Avoid overflowing totals
 	data.MicroAlgos.Raw = crypto.RandUint64() % (1 << 32)
+	// 0 is an invalid round, but would be right if never proposed
+	data.LastProposed = basics.Round(crypto.RandUint64() % 10)
+	// 0 is an invalid round, but would be right if never needed a heartbeat
+	data.LastHeartbeat = basics.Round(crypto.RandUint64() % 10)
 
 	switch crypto.RandUint64() % 3 {
 	case 0:
 		data.Status = basics.Online
-		data.VoteLastValid = 10000
+		data.IncentiveEligible = crypto.RandUint64()%5 == 0
 	case 1:
 		data.Status = basics.Offline
-		data.VoteLastValid = 0
-	default:
+	case 2:
 		data.Status = basics.NotParticipating
 	}
 
-	data.VoteFirstValid = 0
+	// Give online accounts voting data, and some of the offline too. They are "suspended".
+	if data.Status == basics.Online || (data.Status == basics.Offline && crypto.RandUint64()%5 == 1) {
+		crypto.RandBytes(data.VoteID[:])
+		crypto.RandBytes(data.SelectionID[:])
+		crypto.RandBytes(data.StateProofID[:])
+		data.VoteFirstValid = basics.Round(crypto.RandUint64())
+		data.VoteLastValid = basics.Round(crypto.RandUint63()) // int64 is the max sqlite can store
+		data.VoteKeyDilution = crypto.RandUint64()
+	}
+
 	data.RewardsBase = rewardsBase
 	return data
 }
 
 // RandomOnlineAccountData is similar to RandomAccountData but always creates online account
 func RandomOnlineAccountData(rewardsBase uint64) basics.AccountData {
-	var data basics.AccountData
-	data.MicroAlgos.Raw = crypto.RandUint64() % (1 << 32)
-	data.Status = basics.Online
-	data.VoteLastValid = 1000
-	data.VoteFirstValid = 0
-	data.RewardsBase = rewardsBase
-	return data
+	for {
+		data := RandomAccountData(rewardsBase)
```

Comment:
I agree the spinning is a bit weird, but I don't follow your suggestion.  I could only come up with two ways to change it, which I didn't like.
1. Just set `Status = basics.Online` sometimes. I didn't like this, because it is not a valid account state if it was made without key material, and we flip it to `Online`.
2. Sometimes, set `Online` and _also_ create all the stuff to make that legal.  I didn't like this because it would mean that this function needs to understand what it takes to make an account validly online. I preferred to leave that knowledge in `RandomAccountData`.